modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFileBootstrap.pm

```diff
@@ -0,0 +1,226 @@
+Copyright [2016-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+ Bio::EnsEMBL::Production::Pipeline::Webdatafile::WebdataFileBootstrap;
+
+=head1 DESCRIPTION
+  Compute bootstrap step for webdatafile dumps
+
+=cut
+
+package Bio::EnsEMBL::Production::Pipeline::Webdatafile::WebdataFileBootstrap;
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Registry;
+use base qw/Bio::EnsEMBL::Production::Pipeline::Common::Base/;
+use Path::Tiny qw(path);
+use Array::Utils qw(intersect);
+use Path::Tiny qw(path);
+use Carp qw/croak/;
+use JSON qw/decode_json/;
+use Bio::EnsEMBL::Production::Pipeline::Webdatafile::lib::GenomeLookup;
+sub param_defaults {
+  my ($self) = @_;
+  return {
+    %{$self->SUPER::param_defaults},
+  };
+}
+
+
+sub run {
+
+  my ($self) = @_;
+  my $species = $self->param('species');
+  my $current_step = $self->param('current_step') ;
+  my $output = $self->param('output_path');
+  my $app_path = $self->param('app_path'); 
+  my $genome_data = {
+    dbname     => $self->param('dbname'),
+    gca        => $self->param('gca'),
+    genome_id  => $self->param('genome_id'),
+    species    => $self->param('species'),
+    version    => $self->param('version'),
+    type       => $self->param('division'),  
+    root_path  => path($self->param('root_path'))
+  };
+
+  my $dba = Bio::EnsEMBL::Registry->get_DBAdaptor( $self->param('species'), $self->param('group') );
+  my $lookup = Bio::EnsEMBL::Production::Pipeline::Webdatafile::lib::GenomeLookup->new("genome_data" => $genome_data); 
+  my $genome = $lookup->get_genome('1');
+
+  #bootstrap
+  $self->get_assembly_report($genome, $dba);
+  $self->report_to_chrom_lookups($genome);
+  $self->generate_genome_summary_bed($genome);
+}
+
+sub get_assembly_report {
+
+  my ($self, $genome, $dba) = @_;
+
+  my $target_path = $genome->genome_report_path()->stringify;
+  open(my $fh, ">", $target_path);
+
+  my $statement = $self->prepare_sql($genome->species_id());
+
+  my $query_result = $dba->dbc->sql_helper()->execute( -SQL => $self->prepare_sql(1) ); 
+  $self->write_column_headers($fh);
+  $self->write_report($fh, $query_result);
+}
+
+sub report_to_chrom_lookups{
+
+   my ($self, $genome) = @_;
+   my $genome_report = $genome->get_genome_report();
+   my $write_seqs_out = (exists $ENV{ENS_WRITE_SEQS}) ? $ENV{ENS_WRITE_SEQS} : 1; # its temporary one
```

Comment:
I think it's better to have all variables defined in the pipeconfig, and passed as necessary; using environment variables is less robust, because it's easy to be unaware of them, or to forget to defined them.

Reply:
will add to pipeline config